lib/msf/core/modules/loader/base.rb

```diff
@@ -469,17 +469,9 @@ def module_path(parent_path, type, module_reference_name)
   #                   the path is not hidden (starts with '.')
   # @return [false] otherwise
   def module_path?(path)
-    module_path = false
-
-    extension = File.extname(path)
-
-    unless (path[0,1] == "." or
-            extension != MODULE_EXTENSION or
-            path =~ UNIT_TEST_REGEX)
-      module_path = true
-    end
-
-    module_path
+    !(path.starts_with?(".") ||
+      !path.ends_with?(MODULE_EXTENSION) ||
+      path =~ UNIT_TEST_REGEX)
```

Comment:
Probably even better performance with:
```suggestion
      path.match?(UNIT_TEST_REGEX))
```

Reply:
Sure (though TBH that didn't show up very high on the profiler list of hot spots)